src/ua/griddynamics/Game.java

```diff
@@ -0,0 +1,183 @@
+            sb.append(System.lineSeparator());
+            rowCounter++;
+        }
+        sb.append("-|---------|");
+        return sb.toString();
+    }
+
+    private int countSurround(int i, int j) {
+        int count = 0;
+        if (i + 1 != HEIGHT && hiddenField[i + 1][j] == X)
+            count++;
+        if (i + 1 != HEIGHT && j + 1 != WIDTH && hiddenField[i + 1][j + 1] == X)
+            count++;
+        if (i - 1 != -1 && j + 1 != WIDTH && hiddenField[i - 1][j + 1] == X)
+            count++;
+        if (i + 1 != HEIGHT && j - 1 != -1 && hiddenField[i + 1][j - 1] == X)
+            count++;
+        if (i - 1 != -1 && j - 1 != -1 && hiddenField[i - 1][j - 1] == X)
+            count++;
+        if (i - 1 != -1 && hiddenField[i - 1][j] == X)
+            count++;
+        if (j + 1 != WIDTH && hiddenField[i][j + 1] == X)
+            count++;
+        if (j - 1 != -1 && hiddenField[i][j - 1] == X)
+            count++;
+        return count;
+    }
+
+    public void numberGenerator() {
+        for (int i = 0; i < HEIGHT; i++) {
+            for (int j = 0; j < WIDTH; j++) {
+                if (hiddenField[i][j] != X) {
+                    hiddenField[i][j] = (char) (countSurround(i, j) + 48);
+                }
+                if (hiddenField[i][j] == '0') {
+                    hiddenField[i][j] = POINT;
+                }
+            }
+        }
+    }
+
+    public void floodFill(int x, int y, boolean[][] visited) {
+        if (x < 0 || x >= 9 || y < 0 || y >= 9) return;
+
+        if (visited[y][x]) return;
+        visited[y][x] = true;
+
+        if (hiddenField[y][x] == X)
+            return;
+
+        if (hiddenField[y][x] == POINT) {
+            visibleField[y][x] = FREE;
+        } else {
+            visibleField[y][x] = hiddenField[y][x];
+            return;
+        }
+
+        if (hiddenField[y][x] != POINT)
+            return;
+
+        floodFill(x + 1, y, visited);
+        floodFill(x - 1, y, visited);
+        floodFill(x, y - 1, visited);
+        floodFill(x, y + 1, visited);
+        floodFill(x - 1, y - 1, visited);
+        floodFill(x - 1, y + 1, visited);
+        floodFill(x + 1, y - 1, visited);
+        floodFill(x + 1, y + 1, visited);
+    }
+
+    public void putAnswer(Coordinate coordinate) {
+        int x = coordinate.getX();
+        int y = coordinate.getY();
+        char mark = coordinate.getMark();
+
+        if (mark == FREE) {
+            if (hiddenField[y][x] == POINT) {
+                visibleField[y][x] = FREE;
+            } else {
+                visibleField[y][x] = hiddenField[y][x];
+            }
+            floodFill(x, y, new boolean[HEIGHT][WIDTH]);
+        }
+        if (mark == MINE) {
+            if (visibleField[y][x] == MINE) {
+                visibleField[y][x] = POINT;
+            } else {
+                visibleField[y][x] = MINE;
+            }
+        }
+    }
+
+    public boolean isFail(Coordinate coordinate) {
```

Comment:
putAnswer(...) can return false if we failed
In this case we don't need a separate function